phone/index.html

```diff
@@ -0,0 +1,167 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <script src="../components/web-animations-js/web-animations.min.js"></script>
+  <link href="styles.css" rel="stylesheet" type="text/css" />
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+</head>
+<body>
+
+<div id="phone">
+  <div class="icons" id="main">
+  </div>
+  <div class="icons" id="cover">
+  </div>
+</div>
+
+<script>
+
+function randomInt(low, high) {
+  return Math.floor(Math.random() * (high - low)) + low;
+}
+
+// Helper to create: <div class="ball"><i class="material-icons">icon_name</i></div>
+var icons = 'cake star thumb_up favorite new_releases comment cast filter_hdr'.split(' ');
+function createBall(opt_icon, opt_color) {
+  var ball = document.createElement('div');
+  ball.className = 'ball';
+  var icon = opt_icon || icons[randomInt(0, icons.length)];
+  ball.innerHTML = '<i class="material-icons">' + icon + '</i>';
+  ball.style.background = opt_color || 'hsl(' + randomInt(200, 360) + ', 100%, 50%)';
+  return ball;
+}
+
+// Create icons on the home screen.
+var total = randomInt(4, 12);
+for (var i = 0; i < total; ++i) {
+  main.appendChild(createBall());
+}
+
+// Create icons in the toolbar/bottom bar.
+cover.appendChild(createBall('phone', 'green'));
+for (var i = 0; i < 2; ++i) {
+  cover.appendChild(createBall());
+}
+
+// Controls the duration and sensitivity of the drag.
+var screenHeight = 480;
+var duration = 1200;
+
+// Creates an array of players that represent a phone toolbar moving upwards. Moves the toolbar as
+// well as the icons you see on the home screen.
+function createPlayers(callbackWhenDone) {
+  var timing = {duration: duration, fill: 'forwards', easing: 'ease-in-out'};
+  var firstPlayer = cover.animate([
+    {transform: 'none'},
+    {transform: 'translateY(-80%)'},
+  ], timing);
+
+  // When the player finishes, it will either be going forwards (toolbar to top), or backwards
+  // (when it is returned to the base).
+  firstPlayer.onfinish = function() {
+    if (firstPlayer.currentTime <= 0) {
+      // The player is going backwards, cancel all players, return to non-animated state.
+      players.forEach(function(player) { player.cancel(); });
+      callbackWhenDone();
+    }
+  };
+
+  // Animate each ball on the main screen moving out in a random direction.
+  timing.easing = 'ease-in';
+  var players = [].slice.call(main.querySelectorAll('.ball')).map(function(ball) {
+    var x = randomInt(-100, +100);
+    var y = randomInt(-400, -200);
+    var deg = randomInt(-25, +25);
```

Comment:
As discussed using random is a bit of a contrived way to show dynamic animation creation. Making the direction based on the relative X coordinate of the mouse/touch event might be a little clearer.